### PR TITLE
Update docs related to React 18

### PR DESCRIPTION
**Pull request by shuding**
- Streaming (concurrent features or the Fizz architecture) is no longer opt-in or alpha, when you have React 18 installed. It's always enabled.
- Switchable Runtime is a new alpha feature (which requires React 18 too, hence it's added under that list).

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: docs/advanced-features/react-18/overview.md [hunk 1]
```diff
@@ -15,14 +15,20 @@ npm install next@latest react@latest react-dom@latest
...
 
 ## React Server Components (Alpha)
 
 Server Components are a new feature in React that let you reduce your JavaScript bundle size by separating server and client-side code. Server Components allow developers to build apps that span the server and client, combining the rich interactivity of client-side apps with the improved performance of traditional server rendering.
 
 Server Components are still in research and development. [Learn how to try Server Components](/docs/advanced-features/react-18/server-components.md) as an experimental feature in Next.js.
+
+## Switchable Runtime (Alpha)
+
+Next.js supports changing the runtime of your application between Node.js and [Edge Runtime](/docs/api-reference/edge-runtime), in a per-page level. For example, you can selectively configure some specific pages to be server-side rendered in the Edge Runtime.
```
Comment (styfle) [inline docs/advanced-features/react-18/overview.md:32]:
```suggestion
Next.js supports changing the runtime of your application between Node.js and [Edge Runtime](/docs/api-reference/edge-runtime.md), at the page level. For example, you can selectively configure specific pages to be server-side rendered in the Edge Runtime.
```

File: docs/advanced-features/react-18/streaming.md [hunk 2]
```diff
@@ -1,24 +1,15 @@
...
+It's worth noting that the experimental React Server Components feature, is based on streaming. You can read more about server components related streaming APIs in [`next/streaming`](/docs/api-reference/next/streaming.md). However, this guide focuses on basic React 18 streaming.
 
-## Enable Streaming SSR
+## Use Streaming SSR
 
-Enabling streaming SSR means React renders your components into streams and the client continues receiving updates from these streams even after the initial SSR response is sent. In other words, when any suspended components resolve down the line, they are rendered on the server and streamed to the client. With this strategy, the app can start emitting HTML even before all the data is ready, improving your app's loading performance. As an added bonus, in streaming SSR mode, the client will also use selective hydration strategy to prioritize component hydration which based on user interaction.
+When you have Suspense in use in a SSR page, there is no extra configuration required to use streaming SSR.
 
-To enable streaming SSR, set the experimental option `runtime` to either `'nodejs'` or `'edge'`:
```
Comment (styfle) [inline docs/advanced-features/react-18/streaming.md:10]: The user doesn't need to use the Edge runtime? What happens if they use the Node.js runtime? Do they still get streaming?

File: docs/advanced-features/react-18/overview.md [hunk 3]
```diff
@@ -15,14 +15,20 @@ npm install next@latest react@latest react-dom@latest
 
 You can now start using React 18's new APIs like `startTransition` and `Suspense` in Next.js.
 
-## Streaming SSR (Alpha)
+## Streaming SSR
 
-Streaming server-rendering (SSR) is an experimental feature in Next.js 12. When enabled, SSR will use the same [Edge Runtime](/docs/api-reference/edge-runtime.md) as [Middleware](/docs/middleware.md).
+Next.js supports React 18's Streaming server-rendering (SSR) out of the box for SSR pages.
```
Comment (leerob) [inline docs/advanced-features/react-18/overview.md:20]:
```suggestion
Next.js supports React 18 streaming server-rendering (SSR) out of the box.
```

File: docs/advanced-features/react-18/overview.md [hunk 4]
```diff
@@ -15,14 +15,20 @@ npm install next@latest react@latest react-dom@latest
...
 
 Server Components are a new feature in React that let you reduce your JavaScript bundle size by separating server and client-side code. Server Components allow developers to build apps that span the server and client, combining the rich interactivity of client-side apps with the improved performance of traditional server rendering.
 
 Server Components are still in research and development. [Learn how to try Server Components](/docs/advanced-features/react-18/server-components.md) as an experimental feature in Next.js.
+
+## Switchable Runtime (Alpha)
+
+Next.js supports changing the runtime of your application between Node.js and [Edge Runtime](/docs/api-reference/edge-runtime), in a per-page level. For example, you can selectively configure some specific pages to be server-side rendered in the Edge Runtime.
+
+This feature is still experimental. [Learn more about switchable runtime](/docs/advanced-features/react-18/switchable-runtime.md).
```
Comment (leerob) [inline docs/advanced-features/react-18/overview.md:34]:
```suggestion
This feature is still experimental. [Learn more about the switchable runtime](/docs/advanced-features/react-18/switchable-runtime.md).
```

File: docs/advanced-features/react-18/streaming.md [hunk 5]
```diff
@@ -1,24 +1,15 @@
-# Streaming SSR (Alpha)
+# Streaming SSR
 
-React 18 will include architectural improvements to React server-side rendering (SSR) performance. This means you can use `Suspense` in your React components in streaming SSR mode and React will render them on the server and send them through HTTP streams.
-It's worth noting that another experimental feature, React Server Components, is based on streaming. You can read more about server components related streaming APIs in [`next/streaming`](/docs/api-reference/next/streaming.md). However, this guide focuses on basic React 18 streaming.
+React 18 includes architectural improvements to React server-side rendering (SSR) performance. This means you can use `Suspense` in your React components in streaming SSR mode and React will render them on the server and send them through HTTP streams.
```
Comment (leerob) [inline docs/advanced-features/react-18/streaming.md:3]:
```suggestion
React 18 includes architectural improvements to React server-side rendering (SSR) performance. This means you can use `Suspense` in your React components in streaming SSR mode and React will render content on the server and send updates through HTTP streams.
```

File: docs/advanced-features/react-18/streaming.md [hunk 6]
```diff
@@ -1,24 +1,15 @@
...
+It's worth noting that the experimental React Server Components feature, is based on streaming. You can read more about server components related streaming APIs in [`next/streaming`](/docs/api-reference/next/streaming.md). However, this guide focuses on basic React 18 streaming.
```
Comment (leerob) [inline docs/advanced-features/react-18/streaming.md:4]:
```suggestion
React Server Components, an experimental feature, is based on streaming. You can read more about Server Components related streaming APIs in [`next/streaming`](/docs/api-reference/next/streaming.md). However, this guide focuses on streaming with React 18.
```

File: docs/advanced-features/react-18/streaming.md [hunk 7]
```diff
@@ -1,24 +1,15 @@
...
 
-## Enable Streaming SSR
+## Use Streaming SSR
```
Comment (leerob) [inline docs/advanced-features/react-18/streaming.md:6]:
```suggestion
## Using Streaming Server-Rendering
```

File: docs/advanced-features/react-18/switchable-runtime.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
+# Switchable Runtime (Alpha)
+
+By default, Next.js uses Node.js as the runtime for page rendering, including pre-rendering, server-side rendering.
+
+If you have [React 18](/docs/advanced-features/react-18/overview) installed, there is a new experimental feature that lets you switch the page runtime between Node.js and [Edge Runtime](/docs/api-reference/edge-runtime). It affects [SSR streaming](/docs/advanced-features/react-18/streaming) and [Server Components](/docs/advanced-features/react-18/server-components) too.
```
Comment (leerob) [inline docs/advanced-features/react-18/switchable-runtime.md:5]:
```suggestion
If you have [React 18](/docs/advanced-features/react-18/overview) installed, there is a new experimental feature that lets you switch the page runtime between Node.js and the [Edge Runtime](/docs/api-reference/edge-runtime). Changing the runtime affects [SSR streaming](/docs/advanced-features/react-18/streaming) and [Server Components](/docs/advanced-features/react-18/server-components) features, as well.
```

File: docs/advanced-features/react-18/switchable-runtime.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+```jsx
+// next.config.js
+module.exports = {
+  experimental: {
+    runtime: 'nodejs',
+  },
+}
+```
+
+This option determines which runtime should be used as the default page rendering runtime.
```
Comment (leerob) [inline docs/advanced-features/react-18/switchable-runtime.md:20]:
```suggestion
This option determines which runtime should be used as the default rendering runtime for all pages.
```

File: docs/advanced-features/react-18/switchable-runtime.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+## Per-page Runtime Option
+
+On each page, it's also optional to export a `runtime` config set to either `'nodejs'` or `'edge'`:
```
Comment (leerob) [inline docs/advanced-features/react-18/switchable-runtime.md:24]:
```suggestion
On each page, you can optionally export a `runtime` config set to either `'nodejs'` or `'edge'`:
```

File: docs/advanced-features/react-18/switchable-runtime.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+```jsx
+export const config = {
+  runtime: 'nodejs',
+}
+```
+
+When both are set, the per-page runtime option overrides the global runtime option. If the per-page runtime is not set, the global runtime option will be used.
```
Comment (leerob) [inline docs/advanced-features/react-18/switchable-runtime.md:32]:
```suggestion
When both the per-page runtime and global runtime are set, the per-page runtime overrides the global runtime. If the per-page runtime is _not_ set, the global runtime option will be used.
```

File: docs/advanced-features/react-18/streaming.md [hunk 12]
```diff
@@ -1,24 +1,15 @@
...
+When you have Suspense in use in a SSR page, there is no extra configuration required to use streaming SSR.
 
-To enable streaming SSR, set the experimental option `runtime` to either `'nodejs'` or `'edge'`:
+All SSR pages have the ability to render components into streams and the client continues receiving updates from these streams even after the initial SSR response is sent. In other words, when any suspended components resolve down the line, they are rendered on the server and streamed to the client. With this strategy, the app can start emitting HTML even before all the data is ready, improving your app's loading performance. As an added bonus, in streaming SSR mode, the client will also use selective hydration strategy to prioritize component hydration which based on user interaction.
```
Comment (leerob) [inline docs/advanced-features/react-18/streaming.md:10]:
```suggestion
All SSR pages have the ability to render components into streams and the client continues receiving updates from these streams even after the initial SSR response is sent. When any suspended components resolve down the line, they are rendered on the server and streamed to the client. This means applications can start emitting HTML even _before_ all the data is ready, improving your app's loading performance.

As an added bonus, in streaming SSR mode the client will also use selective hydration to prioritize component hydration based on user interactions, further improving performance.
```

File: docs/advanced-features/react-18/streaming.md [hunk 13]
```diff
@@ -1,24 +1,15 @@
...
+It's worth noting that the experimental React Server Components feature, is based on streaming. You can read more about server components related streaming APIs in [`next/streaming`](/docs/api-reference/next/streaming.md). However, this guide focuses on basic React 18 streaming.
 
-## Enable Streaming SSR
+## Use Streaming SSR
 
-Enabling streaming SSR means React renders your components into streams and the client continues receiving updates from these streams even after the initial SSR response is sent. In other words, when any suspended components resolve down the line, they are rendered on the server and streamed to the client. With this strategy, the app can start emitting HTML even before all the data is ready, improving your app's loading performance. As an added bonus, in streaming SSR mode, the client will also use selective hydration strategy to prioritize component hydration which based on user interaction.
+When you have Suspense in use in a SSR page, there is no extra configuration required to use streaming SSR.
```
Comment (leerob) [inline docs/advanced-features/react-18/streaming.md:8]:
```suggestion
When you use Suspense in a server-rendered page, there is no extra configuration required to use streaming SSR.
```

File: docs/advanced-features/react-18/streaming.md (hunk 2 above)
Comment (leerob) [reply docs/advanced-features/react-18/streaming.md:10]: We should probably clarify this - you could stream using the Node.js runtime if you were self-hosting.

Comment (shuding) [reply docs/advanced-features/react-18/streaming.md:10]: No they don't need to "enable" streaming anymore, the rendering architecture is always enabled now. If they use the Edge runtime, they get streaming from the Edge. If they use the Node.js runtime, they will have streaming if Next.js is running in a Node.js server. And if it's running in a lambda function that doesn't support streaming, the response will be buffered and sent all at once when everything is ready.

File: docs/advanced-features/react-18/streaming.md [hunk 14]
```diff
@@ -1,24 +1,17 @@
...
+React Server Components, an experimental feature, is based on streaming. You can read more about Server Components related streaming APIs in [`next/streaming`](/docs/api-reference/next/streaming.md). However, this guide focuses on streaming with React 18.
 
-## Enable Streaming SSR
+## Using Streaming Server-Rendering
 
-Enabling streaming SSR means React renders your components into streams and the client continues receiving updates from these streams even after the initial SSR response is sent. In other words, when any suspended components resolve down the line, they are rendered on the server and streamed to the client. With this strategy, the app can start emitting HTML even before all the data is ready, improving your app's loading performance. As an added bonus, in streaming SSR mode, the client will also use selective hydration strategy to prioritize component hydration which based on user interaction.
+When you use Suspense in a server-rendered page, there is no extra configuration required to use streaming SSR.
```
Comment (leerob) [inline docs/advanced-features/react-18/streaming.md:8]:
```suggestion
When you use Suspense in a server-rendered page, there is no extra configuration required to use streaming SSR. When deployed, streaming can be utilized through infrastructure like [Edge Functions](https://vercel.com/edge) on Vercel (with the Edge Runtime) or with a Node.js server (with the Node.js runtime). AWS Lambda Functions do not currently support streaming responses.
```

**Review by leerob — APPROVED** [review]

File: docs/advanced-features/react-18/switchable-runtime.md [hunk 15]
```diff
@@ -0,0 +1,34 @@
+# Switchable Runtime (Alpha)
+
+By default, Next.js uses Node.js as the runtime for page rendering, including pre-rendering, server-side rendering.
```
Comment (styfle) [inline docs/advanced-features/react-18/switchable-runtime.md:3]:
```suggestion
By default, Next.js uses Node.js as the runtime for page rendering, including pre-rendering and server-side rendering.
```

File: docs/advanced-features/react-18/switchable-runtime.md [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+module.exports = {
+  experimental: {
+    runtime: 'nodejs',
+  },
+}
+```
+
+This option determines which runtime should be used as the default rendering runtime for all pages.
+
+## Per-page Runtime Option
```
Comment (styfle) [inline docs/advanced-features/react-18/switchable-runtime.md:22]:
```suggestion
## Page Runtime Option
```

File: docs/advanced-features/react-18/switchable-runtime.md [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+
+On each page, you can optionally export a `runtime` config set to either `'nodejs'` or `'edge'`:
+
+```jsx
+export const config = {
+  runtime: 'nodejs',
+}
+```
```
Comment (styfle) [inline docs/advanced-features/react-18/switchable-runtime.md:30]: Does this also work for `/pages/api/**`?

**Pull request merged**